# A. Background

In this section, we will predict delay probability of a data from one shipment. The output will predict the probability of delay in the shipment which will be analyze from some features given.

# B. Import Libraries

In [ ]:
import pickle
import pandas as pd
import numpy as np

# C. Loading Model

In [ ]:
with open('model_fix.pkl', 'rb') as file:
    model_fix = pickle.load(file)

# D. Loading Data

In [ ]:
#entry data 
shipment_1 = {
    'fuel_consumption_rate' : 13.005,
    'eta_variation_hours' : 2.89,
    'traffic_congestion_level' : 8.76,
    'warehouse_inventory_level' : 103.55,
    'loading_unloading_time' : 4.67,
    'handling_equipment_availability' : 0.35,
    'order_fulfillment_status' : 0.43,
    'weather_condition_severity' : 0.65,
    'port_congestion_level' : 5.65,
    'shipping_costs' : 230.77,
    'supplier_reliability_score' : 0.59,
    'lead_time_days' : 1.34,
    'historical_demand' : 2198.33,
    'iot_temperature' : 0.55,
    'cargo_condition_status' : 0.87,
    'route_risk_level' : 7.23,
    'customs_clearance_time' : 3.76,
    'driver_behavior_score' : 0.98,
    'fatigue_monitoring_score' : 0.68,
    'month' : 9,
    'year' : 2024 
    }
shipment_1 = pd.DataFrame([shipment_1])
shipment_1

fuel_consumption_rate  eta_variation_hours  traffic_congestion_level  \
0                 13.005                 2.89                      8.76   

   warehouse_inventory_level  loading_unloading_time  \
0                     103.55                    4.67   

   handling_equipment_availability  order_fulfillment_status  \
0                             0.35                      0.43   

   weather_condition_severity  port_congestion_level  shipping_costs  ...  \
0                        0.65                   5.65          230.77  ...   

   lead_time_days  historical_demand  iot_temperature  cargo_condition_status  \
0            1.34            2198.33             0.55                    0.87   

   route_risk_level  customs_clearance_time  driver_behavior_score  \
0              7.23                    3.76                   0.98   

   fatigue_monitoring_score  month  year  
0                      0.68      9  2024  

[1 rows x 21 columns]

# E. Model Inference

In [ ]:
# Prediksi User 1
pred_delay_1 = model_fix.predict(shipment_1)
print(pred_delay_1)


[0.67331139]


# F. Conclusion

Based on model prediction, the delay probability is 0.67. This means that the is shipment_1 is classified as **Moderate Risk** in the delay probability. 

However, since MAE score is 25-28% so it means that actual probability might be 0.37 - 0.95 for shipment_1.